Sources/SwiftLexicalLookup/IdentifiableSyntax.swift

```diff
@@ -0,0 +1,38 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+/// Syntax node that can be refered to with an identifier.
+public protocol IdentifiableSyntax: SyntaxProtocol {
```

Comment:
Let's iterate with this protocol some more. We might want to sink it down into SwiftSyntax and make the NamedDeclSyntax protocol inherit from it.

Reply:
(this doesn't have to happen now)

Reply:
I'll keep adding the protocol to more syntax nodes like `FunctionParameterSyntax` and `GenericParameterSyntax` in the coming PRs. Also, would it be a good idea to soon fix the problem with `ClosureCaptureSyntax ` so we can complete implementation of `IdentifiableSyntax`?

Reply:
I think the fix with `ClosureCaptureSyntax` should be a separate PR from this one entirely, since it will involve changing the syntax tree. I don't specifically mind whether it goes before or after this PR; we can fix it up either way. (But we shouldn't sink this protocol down into SwiftSyntax until we fix the issue)